src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -213,7 +228,16 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
 
             using (OpenNewDocumentStateScope(options ?? workspace.Options))
             {
-                var document = OpenDocument(workspace, documentId);
+                // Before attempting to open the document, check if the location maps to a different file that should be opened instead.
+                var document = workspace.CurrentSolution.GetDocument(documentId);
+                var spanMappingService = document?.Services.GetService<ISpanMappingService>();
+                if (spanMappingService != null)
```

Comment:
Should this be checking for document being null and bailing right away? That might cleanup the the other null checking after the call to OpenDocument.

Reply:
I actually can't - OpenDocument can return null if the document doesn't open